context.go

```diff
@@ -16,14 +16,41 @@ import (
 	"github.com/sdboyer/gps"
 )
 
+type BuildContext interface {
+	SourceManager() (gps.SourceManager, error)
+
+	// LoadProject takes a path and searches up the directory tree for a project
+	// root. If an absolute path is given, the search begins in that directory. If
+	// a relative or empty path is given, the search start is computed from the
+	// current working directory. The search stops when a file with the name
+	// ManifestName (manifest.json, by default) is located.
+	//
+	// The Project contains the parsed manifest as well as a parsed lock file, if
+	// present. The import path is calculated as the remaining path segment below
+	// Ctx.GOPATH/src.
+	LoadProject(path string) (*Project, error)
+
+	// SplitAbsoluteProjectRoot takes an absolute path and compares it against
+	// declared GOPATH(s) to determine what portion of the input path should be
+	// treated as an import path - as a project root.
+	//
+	// The second returned string indicates which GOPATH value was used.
+	SplitAbsoluteProjectRoot(path string) (string, error)
+
+	VersionInWorkspace(root gps.ProjectRoot) (gps.Version, error)
+
+	// Returns the root directory where packages can be found.
+	Root() string
```

Comment:
Yep, I agree. I was trying to do the minimal change for the first set to submit for PR.

I think some sort of search-gopath-for-a-project method seems right.